src/datepicker/datepicker.ts

```diff
@@ -298,6 +331,30 @@ export class NgbDatepicker implements OnDestroy,
     });
   }
 
+  /**
+   *  Returns a copy of the state of the datepicker.
+   */
+  get state(): NgbDatepickerState {
+    Object.assign(this.publicState, {
```

Comment:
You're still kind of creating a new object every time change detection runs with `datepicker.state`, ex in 

```html
 {{ datepicker.state.maxDate }}
```